test-data/unit/check-callable.test

```diff
@@ -208,13 +208,13 @@ b = B()  # type: B
 c = A()  # type: Union[A, B]
 
 if callable(a):
-    5 + 'test'
+    5 + 'test' # E: Unsupported operand types for + ("int" and "str")
 
 if not callable(b):
     5 + 'test'
 
 if callable(c):
-    reveal_type(c)  # E: Revealed type is '__main__.B'
+    reveal_type(c)  # E: Revealed type is 'Union[<callable subtype of A>, __main__.B]'
```

Comment:
Just a random comment: we need to be sure that this type ``<callable subtype of A>`` will never appear in error messages (apart from actually ``reveal_type``), since this can be confusing. I think it can be formatted just as ``A`` in other cases

Reply:
Does there exist mechanism for having different internal and external names for instances?

Reply:
Yes, see ``messages.py``. There is a method ``MessageBuilder.format`` that controls most of the formatting. _But_, see other comments for this code that propose more radical solutions.